text/0059-token-metadata-document.md

```diff
@@ -0,0 +1,275 @@
+
+We take the point of view that these differences are not "token metadata", which are determined by the ID of the token that we are using, but rather localization properties determined by the locale of the user. So, for example the same app might be capable of displaying `kr 127,54` and `127.54 kr`, depending on the locale of the user, and token metadata does not influence this.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+The token-metadata json fields have the following semantics:
+
+* `schema_version`: An integer describing the version number of the schema of this document. This is 1 until a future MCIP proposes schema modifications.
+* `signing_timestamp`: A string representation of the unix timestamp at the (approximate) time of creating the document's signature. This is an integer number of seconds since the unix epoch.
+* `token_id`: A decimal string representation of the `u64` corresponding to the token id.
+* `currency_name`: A UTF-8 string which is the full, official name of the currency. This is intended to be an English language noun or noun phrase.
+* `short_code`: A short code for the currency, up to 12 ASCII characters matching `/[A-Za-z0-9.-]{1,12}/`. This may be displayed to users in a manner similar to ISO 4217 currency codes, for example, USD, GBP, CAD, in connection to an amount of currency. This is expected to match the ticker symbol used for this asset on cryptocurrency exchanges.
+* `symbol`: Optional UTF-8 string. Most fiat currencies have a printable character such as $, £, ¥. Some cryptocurrencies do also. Bitcoin has ₿. Ethereum has Ξ.
+* `decimals`: An integer specifying how the `u64` integer in a TxOut in the blockchain is scaled to compute a user-displayed amount of the currency. For example, MOB has 12 decimals, which indicates that `10^12` of the smallest representable units of MOB on the MobileCoin blockchain are equal to one MOB.
+* `logo_svg`: An optional logo image. This is a base64-encoded SVG document. This is expected to have been sanitized using something like [svg-hush](https://github.com/cloudflare/svg-hush), to remove scripting, hyperlinks to other documents, and references to cross-origin resources. The full extent of such sanitization will not be specified here.
+* `info_url`: A link to a website containing more information about the token that may be interesting to token holders. This should have basic information about the purpose of the token, its supply, any utility that it has, or links to associated whitepapers.
+
+Clients MUST download and validate the `token_metadata.sig` against the bytes of the `token_metadata.json` and the `MINTING_TRUST_ROOT` public key before attempting to process the `token_metadata.json`. Clients MUST reject the json with an error if the signature checking fails.
```

Comment:
I expect situations will arise where clients are unaware of this requirement. Creating a helper method that parses the token_metadata.json and returns an error on signature mismatch feels important for meeting this requirement. If the json file is publicly available, it is otherwise likely there will be instances where it is fetched and used raw.